File: backend/src/main/java/com/back/domain/pet/exception/PetException.java
package com.back.domain.pet.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;


@Getter
public class PetException extends RuntimeException {
    private final HttpStatus httpStatus;
    private final String code;
    private final String message;

    public PetException(PetErrorCode memberErrorCode) {
        super(memberErrorCode.getMessage());
        this.httpStatus = memberErrorCode.getHttpStatus();
        this.code = memberErrorCode.getCode();
        this.message = memberErrorCode.getMessage();
    }
}

File: backend/src/main/java/com/back/domain/pet/exception/PetException.kt
package com.back.domain.pet.exception

import org.springframework.http.HttpStatus

class PetException(
    val httpStatus: HttpStatus,
    val code: String,
    val errorMessage: String
) : RuntimeException(errorMessage) {

    constructor(errorCode: PetErrorCode) : this(
        httpStatus = errorCode.httpStatus,
        code = errorCode.code,
        errorMessage = errorCode.message
    )
}